packages/quantic/force-app/main/default/lwc/quanticFoldedResultList/__tests__/quanticFoldedResultList.test.js

```diff
@@ -0,0 +1,340 @@
+let resultsPerPageState = initialResultsPerPageState;
+
+const functionsMocks = {
+  buildFoldedResultList: jest.fn(() => ({
+    subscribe: functionsMocks.subscribeFoldedResultList,
+    state: foldedResultsListState,
+  })),
+  subscribeFoldedResultList: jest.fn((cb) => {
+    cb();
+    return functionsMocks.unsubscribeFoldedResultList;
+  }),
+  unsubscribeFoldedResultList: jest.fn(),
+  buildResultsPerPage: jest.fn(() => ({
+    subscribe: functionsMocks.subscribeResultsPerPage,
+    state: resultsPerPageState,
+  })),
+  subscribeResultsPerPage: jest.fn((cb) => {
+    cb();
+    return functionsMocks.unsubscribeResultsPerPage;
+  }),
+  unsubscribeResultsPerPage: jest.fn(),
+  buildResultTemplatesManager: jest.fn(),
+};
+
+const selectors = {
+  placeholder: 'c-quantic-placeholder',
+  result: 'c-quantic-result',
+  error: 'c-quantic-component-error',
+};
+
+const exampleEngine = {
+  id: defaultOptions.engineId,
+};
+
+const createTestComponent = buildCreateTestComponent(
+  QuanticFoldedResultList,
+  'c-quantic-folded-result-list',
+  defaultOptions
+);
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  mockHeadlessLoader.getHeadlessBundle = () => {
+    return {
+      buildFoldedResultList: functionsMocks.buildFoldedResultList,
+      buildResultsPerPage: functionsMocks.buildResultsPerPage,
+      buildResultTemplatesManager: functionsMocks.buildResultTemplatesManager,
+    };
+  };
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticFoldedResultList && !isInitialized) {
+      isInitialized = true;
+      initialize({
+        id: defaultOptions.engineId,
+      });
+    }
+  };
+}
+
+function mockErroneousHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element) => {
+    if (element instanceof QuanticFoldedResultList) {
+      element.setInitializationError();
+    }
+  };
+}
+
+describe('c-quantic-folded-result-list', () => {
+  beforeEach(() => {
+    mockSuccessfulHeadlessInitialization();
+    prepareHeadlessState();
+  });
+  afterEach(() => {
+    cleanup();
+    foldedResultsListState = initialFoldedResultListState;
+    resultsPerPageState = initialResultsPerPageState;
+    isInitialized = false;
+    jest.clearAllMocks();
+  });
+
+  describe('component initialization', () => {
+    describe('when an initialization error occurs', () => {
+      beforeEach(() => {
+        mockErroneousHeadlessInitialization();
+      });
+
+      it('should display the initialization error component', async () => {
+        const element = createTestComponent();
+        document.body.appendChild(element);
+        await flushPromises();
+        const errorComponent = element.shadowRoot.querySelector(
+          selectors.error
+        );
+        expect(errorComponent).not.toBeNull();
+      });
+    });
+
+    describe('successful initialization', () => {
```

Comment:
I would add tests here to make sure that the `buildResultsPerPage` controller is being built and `buildResultTemplatesManager` controller as well